esp-hal-common/src/rtc_cntl/rtc/esp32c6.rs

```diff
@@ -30,8 +35,1082 @@ const I2C_DIG_REG_SCK_DCAP: u8 = 14;
+    pub u8,   hp_active2sleep_backup_modem_clk_code, set_hp_active2sleep_backup_modem_clk_code: 9, 8;
+    pub bool, hp_sleep_retention_mode              , set_hp_sleep_retention_mode              : 10;
+    pub bool, hp_modem2sleep_retention_en          , set_hp_modem2sleep_retention_en          : 12;
+    pub bool, hp_active2sleep_retention_en         , set_hp_active2sleep_retention_en         : 13;
+    pub u8,   hp_modem2sleep_backup_clk_sel        , set_hp_modem2sleep_backup_clk_sel        : 17, 16;
+    pub u8,   hp_active2sleep_backup_clk_sel       , set_hp_active2sleep_backup_clk_sel       : 19, 18;
+    pub u8,   hp_modem2sleep_backup_mode           , set_hp_modem2sleep_backup_mode           : 25, 23;
+    pub u8,   hp_active2sleep_backup_mode          , set_hp_active2sleep_backup_mode          : 28, 26;
+    pub bool, hp_modem2sleep_backup_en             , set_hp_modem2sleep_backup_en             : 30;
+    pub bool, hp_active2sleep_backup_en            , set_hp_active2sleep_backup_en            : 31;
+}
+
+bitfield::bitfield! {
+    #[derive(Clone, Copy, Default)]
+    // custom based on `PMU_ICG_FUNC_ENA_*` bitflag constants
+    pub struct HpBackupClk(u32);
+
+    pub bool, gdma        , set_gdma         : 0;
+    pub bool, spi2        , set_spi2         : 1;
+    pub bool, i2s_rx      , set_i2s_rx       : 2;
+    pub bool, uart0       , set_uart0        : 3;
+    pub bool, uart1       , set_uart1        : 4;
+    pub bool, uhci        , set_uhci         : 5;
+    pub bool, usb_device  , set_usb_device   : 6;
+    pub bool, i2s_tx      , set_i2s_tx       : 7;
+    pub bool, regdma      , set_regdma       : 8;
+    pub bool, retention   , set_retention    : 9;
+    pub bool, mem_monitor , set_mem_monitor  : 10;
+    pub bool, sdio_slave  , set_sdio_slave   : 11;
+    pub bool, tsens       , set_tsens        : 12;
+    pub bool, tg1         , set_tg1          : 13;
+    pub bool, tg0         , set_tg0          : 14;
+    pub bool, hpbus       , set_hpbus        : 15;
+    pub bool, soc_etm     , set_soc_etm      : 16;
+    pub bool, hpcore      , set_hpcore       : 17;
+    pub bool, systimer    , set_systimer     : 18;
+    pub bool, sec         , set_sec          : 19;
+    pub bool, saradc      , set_saradc       : 20;
+    pub bool, rmt         , set_rmt          : 21;
+    pub bool, pwm         , set_pwm          : 22;
+    pub bool, pvt_monitor , set_pvt_monitor  : 23;
+    pub bool, parl_tx     , set_parl_tx      : 24;
+    pub bool, parl_rx     , set_parl_rx      : 25;
+    pub bool, mspi        , set_mspi         : 26;
+    pub bool, ledc        , set_ledc         : 27;
+    pub bool, iomux       , set_iomux        : 28;
+    pub bool, i2c         , set_i2c          : 29;
+    pub bool, can1        , set_can1         : 30;
+    pub bool, can0        , set_can0         : 31;
+}
+
+macro_rules! hp_system_init {
+    ($state:ident => $s:ident) => {
+        paste::paste! {
+            unsafe {
```

Comment:
I'm sorry for these macros. esp-idf just indexes into a register group with 3 (HP) or 2 (LP) groups. Unfortunately, in the PAC, the register field names are different for each power state(?) (they contain a bunch of the register's name for each field) and transmuting them would just be super confusing.